File: java_basics/src/test/java/com/user/selenium/CheckboxesTest.java
package com.user.selenium;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.annotations.Test;

import java.util.List;

public class CheckboxesTest extends BaseTest {

    @Test
    public void testCheckboxesToggle() {
        logger.info("STARTING Test --- Checkboxes Toggle ---");

        String url = "https://the-internet.herokuapp.com/checkboxes";
        driver.get(url);
        logger.info("Navigated to " + url);

        List<WebElement> checkboxes = driver.findElements(By.cssSelector("input[type='checkbox']"));
        logger.info("Found " + checkboxes.size() + " checkboxes");

        int index = 1;
        for (WebElement box : checkboxes) {
            boolean initialState = box.isSelected();
            logger.info("Checkbox " + index + " initial state: " + initialState);

            // Toggle
            box.click();
            boolean toggledState = box.isSelected();
            logger.info("Checkbox " + index + " toggled state: " + toggledState);

            assert toggledState != initialState :
                    "Checkbox " + index + " did not change state (was " + initialState + ", now " + toggledState + ")";

            index++;
        }

        logger.info("ENDING Test --- Completed successfully: Checkboxes Toggle ---");
    }
}

File: kotlin_basics/src/test/kotlin/tests/CheckboxesTest.kt
package tests

import base.BaseTest
import org.openqa.selenium.By
import org.testng.Assert
import org.testng.annotations.Test

class CheckboxesTest : BaseTest() {

    @Test
    fun testCheckboxesToggle() {
        logger.info("STARTING Test --- Checkboxes Toggle ---")

        val url = "https://the-internet.herokuapp.com/checkboxes"
        driver.get(url)
        logger.info("Navigated to $url")

        val checkboxes = driver.findElements(By.cssSelector("input[type='checkbox']"))
        logger.info("Found ${checkboxes.size} checkboxes")

        for ((index, box) in checkboxes.withIndex()) {
            val checkboxNumber = index + 1

            val initialState = box.isSelected
            logger.info("Checkbox $checkboxNumber initial state: $initialState")

            // Toggle state
            box.click()
            val toggledState = box.isSelected
            logger.info("Checkbox $checkboxNumber toggled state: $toggledState")

            Assert.assertNotEquals(
                toggledState,
                initialState,
                "Checkbox $checkboxNumber did not change state (was $initialState, now $toggledState)"
            )
        }

        logger.info("ENDING Test --- Completed successfully: Checkboxes Toggle ---")
    }
}
